ui/app/adapters/variable.js

```diff
@@ -21,6 +21,12 @@ export default class VariableAdapter extends ApplicationAdapter {
     let baseUrl = this.buildURL(modelName);
     return pluralize(baseUrl);
   }
+
+  urlForQuery(_query, modelName) {
+    let baseUrl = this.buildURL(modelName);
+    return pluralize(baseUrl);
+  }
```

Comment:
Is this method used anywhere?

It seems like perhaps this needs to be implemented further; as it stands, if I save a variable on namespace A, my request for that variable immediately upon creation fails. In practice, this throws a 404 immediately upon creating a new variable with a namespace on this branch. 
![image](https://user-images.githubusercontent.com/713991/178824296-a6df9dfc-294a-43ab-ac97-4ca335914647.png)

Reply:
`urlForQuery` is invoked when we use the `query` method on the `store` service.

Reply:
Specifically, we're handling how we structure the resource in the request URL.

Sometimes we must be plural `/v1/vars/` and other times we must be singular `/v1/var/:resourceId`.

Reply:
This behavior is correct here. However, it might have a collision with [13627](https://github.com/hashicorp/nomad/pull/13627).

Reply:
This seems totally resolved now! 🎉